### DISCUSSION TITLE
Reverse selection - get all tags except one or two?

### DISCUSSION THREAD

**Discussion by euro** [11ty/eleventy#2813] [Uncategorized]
### Describe the solution you'd like

I see you have baked in the ability to create a custom collection of a tag (or tags) using the following:

```
  // Get only content that matches a tag
  eleventyConfig.addCollection("myTravelPostsWithPhotos", function(collectionApi) {
    return collectionApi.getFilteredByTags("post", "travel", "photo");
  });
```

Is there a way to do a negative match? Something like:
```
 eleventyConfig.addCollection("newCollection" = collection.whatever but filter out (!'unwanted')
```

or even easier:

{% for post in collections.all | except('unwanted', 'another one') %}

**Comment by pdehaan** [comment]
The collection API's `.getFilteredByTags()` should return an array of pages, so you can use `Array#filter()` to do what you need.

Something like:
```js
  // Get only content that matches a tag
  eleventyConfig.addCollection("myTravelPostsWithPhotos", function(collectionApi) {
    const posts = [...collectionApi.getFilteredByTags("post", "travel", "photo")];
    return posts.filter(post => !post.data.tags.includes("unwanted"));
  });
```

**Reply by pdehaan** [reply]
Or if you want the custom filter approach, this might get you there:

```js
  eleventyConfig.addFilter("except", function (collection = [], ...unwantedTags) {
    return collection.filter(post => {
      return !unwantedTags.some(tag => post.data.tags?.includes(tag));
    });
  });
```

### USAGE (index.njk)
```njk
{% for post in collections.all | except('unwanted', 'another one') %}
  <p>{{ post.data.title }}</p>
{% endfor %}
```

**Reply by euro** [reply]
I tried the second approach where you created the except filter but it randomly caused an error in the console when saving pages. 

I was able to use the collection API example with the ```!post.data.tags.includes("unwanted")``` and basically hardcoded it in the 11ty config file.

What I was trying to do but I wasn't able to articulate it correctly is a whole other level of crazy. Ive discovered that I need another level of taxonomy and am reading through some blog posts on adding a category key and hacking the system to do what I need.

I was trying to use a set of common tags among several posts and group them a second way which I found a challenge because those common "tags" I was using to organize content really aren't a tag, it's a category of tags.



Thanks for the tip!

**Reply by euro** [reply]
This is what I was looking for: https://github.com/11ty/eleventy/issues/332#issuecomment-445236776
